rest-service/manager_rest/dsl_functions.py

```diff
@@ -32,86 +32,113 @@
+    except parser_exceptions.FunctionEvaluationError, e:
+        raise FunctionsEvaluationError(str(e))
+
+
+def evaluate_node_instance(instance):
+    deployment_id = instance['deployment_id']
+    sm = get_storage_manager()
+    sm.get(Deployment, deployment_id, include=['id'])
+    storage = DBEvaluationStorage(deployment_id, sm)
+
+    try:
+        return functions.evaluate_node_instance_functions(instance, storage)
+    except parser_exceptions.FunctionEvaluationError as e:
+        raise FunctionsEvaluationError(str(e))
+
+
 def evaluate_intrinsic_functions(payload, deployment_id, context=None):
     context = context or {}
     sm = get_storage_manager()
     sm.get(Deployment, deployment_id, include=['id'])
-    methods = _get_methods(deployment_id, sm)
+    storage = DBEvaluationStorage(deployment_id, sm)
 
     try:
-        return functions.evaluate_functions(
-            payload=payload,
-            context=context,
-            **methods)
-    except parser_exceptions.FunctionEvaluationError, e:
+        return functions.evaluate_functions(payload, context, storage)
+    except parser_exceptions.FunctionEvaluationError as e:
         raise FunctionsEvaluationError(str(e))
 
 
 def evaluate_deployment_outputs(deployment_id):
     sm = get_storage_manager()
     deployment = sm.get(Deployment, deployment_id, include=['outputs'])
-    methods = _get_methods(deployment_id, sm)
+    storage = DBEvaluationStorage(deployment_id, sm)
 
     if not deployment.outputs:
         return {}
 
     try:
-        return functions.evaluate_outputs(
-            outputs_def=deployment.outputs,
-            **methods
-            )
-    except parser_exceptions.FunctionEvaluationError, e:
+        return functions.evaluate_outputs(deployment.outputs, storage)
+    except parser_exceptions.FunctionEvaluationError as e:
         raise DeploymentOutputsEvaluationError(str(e))
 
 
 def evaluate_deployment_capabilities(deployment_id):
     sm = get_storage_manager()
     deployment = sm.get(Deployment, deployment_id, include=['capabilities'])
-    methods = _get_methods(deployment_id, sm)
+    storage = DBEvaluationStorage(deployment_id, sm)
 
     if not deployment.capabilities:
         return {}
 
     try:
         return functions.evaluate_capabilities(
-            capabilities=deployment.capabilities,
-            **methods
-            )
+            deployment.capabilities, storage)
     except parser_exceptions.FunctionEvaluationError, e:
         raise DeploymentCapabilitiesEvaluationError(str(e))
 
 
-def get_secret_method():
+def get_secret_method(secret_key):
     sm = get_storage_manager()
-    methods = _get_methods(None, sm)
-    return methods['get_secret_method']
+    secret = sm.get(Secret, secret_key)
+    decrypted_value = cryptography_utils.decrypt(secret.value)
+    return SecretType(secret_key, decrypted_value)
 
 
-def _get_methods(deployment_id, storage_manager):
+class DBEvaluationStorage(object):
```

Comment:
maybe DBEvaluationStorage => EvalutedFunctionsStorage

Reply:
`FunctionEvaluationDBStorage`, it should have `DB` to distinguish from other function evaluation storages (local's FileStorage and InMemoryStorage)